### ISSUE TITLE
Inconsistent behavior of `torch.Tensor.scatter_` for float `src`

### ISSUE THREAD

**Issue by patmjen** [pytorch/pytorch#61854]
## 🐛 Bug
Using `torch.Tensor.scatter_` with float `src`  (i.e. a single number not a `Tensor`) has inconsistent behavior for the CPU and CUDA version. 

## To Reproduce
The snippet below shows the inconsistency.
```python
import torch
device = torch.device('cuda')  # This will work fine
# device = torch.device('cpu')  # This will give an error
x = torch.arange(10, device=device)
idx1 = torch.tensor([4, 2, 0, 8, 0, 1], device=device)
idx2 = torch.tensor([4, 2, 0, 8, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1], device=device)  # Too long for CPU

print('idx1.numel() =', idx1.numel())
print(x.scatter_(0, idx1, 0))
# idx1.numel() = 6
# tensor([0, 0, 0, 3, 0, 5, 6, 7, 0, 9], device='cuda:0')

print('idx2.numel() =', idx2.numel())
print(x.scatter_(0, idx2, 0))
# idx2.numel() = 14
# tensor([0, 0, 0, 3, 0, 5, 6, 7, 0, 9], device='cuda:0')
```
The CUDA version will work and produce the expected output (to my mind), but the CPU version will raise an error message for `idx2`, since it is longer than `x`.

## Expected behavior

I would expect the CPU and CUDA version to have the same behavior - either both producing an error message or both giving the same output.  As I can tell, the documentation specifies that "`index.size(d) <= self.size(d)` for all dimensions `d != dim`" but does not require it for `d == dim`. Therefore, it is not clear to me which one it should be. Having both versions work without error seems ideal, but I of course do not have the full context.

## Environment

```bash
$ python3 collect_env.py
Collecting environment information...
PyTorch version: 1.9.0+cu111
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Scientific Linux release 7.7 (Nitrogen) (x86_64)
GCC version: (GCC) 9.3.1 20210402
Clang version: Could not collect
CMake version: version 2.8.12.2
Libc version: glibc-2.17

Python version: 3.8.9 (default, Apr 13 2021, 20:52:41)  [GCC 9.3.1 20210402] (64-bit runtime)
Python platform: Linux-3.10.0-1160.31.1.el7.x86_64-x86_64-with-glibc2.2.5
Is CUDA available: True
CUDA runtime version: 11.1.74
GPU models and configuration:
GPU 0: Tesla V100-SXM2-32GB
GPU 1: Tesla V100-SXM2-32GB
GPU 2: Tesla V100-SXM2-32GB
GPU 3: Tesla V100-SXM2-32GB

Nvidia driver version: 470.42.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.1
[pip3] torch==1.9.0+cu111
[pip3] torchaudio==0.9.0
[pip3] torchvision==0.10.0+cu111
[conda] Could not collect
```

## Additional context

This inconsistency also causes a bug in PyTorch3D's function `cubify`, as explained [here](https://github.com/facebookresearch/pytorch3d/issues/513).

cc @VitalyFedyunin

**Comment by vadimkantorov** [comment]
Also stumbled on this: https://discuss.pytorch.org/t/scatter-for-computing-histogram-problems-with-dimensions/127278

Also, the error messages for scatter_ are still VERY hard to parse and understand what's wrong and why it's not supported

**Comment by ngimel** [comment]
cc @nikitaved

**Comment by redmagnos** [comment]
So I think this must've been fixed at some point? Running either the original code to reproduce or the snipped linked by @vadimkantorov seems to work just fine with cpu as the device.

**Comment by ngimel** [comment]
Yeah, closing as fixed
